src/data_storage/data_source.py

```diff
@@ -15,3 +22,26 @@ def get_node(self, story_id: str, uri: str) -> Node:
     @abc.abstractmethod
     def get_story_list(self) -> StoryList:
         """Get list of stories"""
+
+    def create_story(self, data: dict):
+        """
+        Template method.
+        Raise exception if unsuccessful.
+        Validate and create story item
+        """
+        validated_story_data = parse_story_structure(data)
+        graph, root_node = parse_graph(validated_story_data)
+        self._validate_story_item(graph, root_node)
+        self._create_story(validated_story_data)
+        return validated_story_data
```

Comment:
I think Data driver should be responsible only for saving story. Parsing/validation we can place in another places.

Reply:
I agree with parsing part with you. In my opinion, validation step is necessary for saving story item. Also, we are free to use the validation function in another place because `def _validate_story_item` is static method

Reply:
Make sense. But `DataDriver._validate_story_item` is not that readable like `validate_story_item` (IMHO). 
Feel free to store `_validate_story_item` where you want :)